extensions/netty/runtime/src/main/java/io/quarkus/netty/runtime/graal/NettySubstitutions.java

```diff
@@ -284,6 +284,37 @@ final ChannelFuture initAndRegister() {
     }
 }
 
+final class Holder_io_netty_util_concurrent_ScheduledFutureTask {
+    private static long START_TIME = Long.MIN_VALUE;
```

Comment:
This `MIN_VALUE` trick is not correct.  In fact there are no special sentinel values possible for `System.nanoTime()`.  The field should be initialized directly in its declaration.  To handle the native image case, the field should be reinitialized in the substitution or the corresponding class should be marked as a runtime-initialization class.

Reply:
Additionally, the field should be `final`.

Reply:
@dmlloyd Yes, there are no sentinel values. However if the first invocation (that should initialize it) gets called when `System.nanoTime()` returns exactly the `Long.MIN_VALUE`, this only means that it gets initialized to a bit higher value sometimes later. Double initialization in this case causes no harm. Another way to get double initialization is when this gets called concurrently - one of the writes eventually wins but generally we don't care.

About the `final` - so if I declare `static final` field here next to the substitution class, will that get really called by the first time when we access the class? Because otherwise we would be back at the original problem when the `System.nanoTime()` value was compiled as constant into the native image.

Reply:
If you declare it final, it will be called when the class is initialized.  In JVM mode, that will be on first access.  In native mode, it's either build time or run time depending on config.  Alternatively, you can create a substitution just for a field that causes just that field to be recomputed at native image start, which overall is a much simpler solution.

Reply:
@dmlloyd Okay, I may have misunderstood `RecomputeFieldValue` javadoc, thinking that it is still recomputed when the image is built - so this computation happens when the native image starts?

Reply:
Correct.  The simplest mode is `FromAlias` where it just reruns the same initialization.  I believe this works by analyzing the compiled `<clinit>` method and tracing the data dependencies of the assignment, possibly using similar logic to the logic that ensures that every static final field is assigned exactly once.

(As you likely know, all static field initializations and static init blocks are rolled into a single, hidden `<clinit>` method by `javac`.)

Reply:
So I've tried to use just this simple substitution:
```java
@TargetClass(className = "io.netty.util.concurrent.ScheduledFutureTask")
final class Target_io_netty_util_concurrent_ScheduledFutureTask {
    @Alias
    @RecomputeFieldValue(kind = RecomputeFieldValue.Kind.FromAlias, isFinal = true)
    @Substitute
    private static long START_TIME = System.nanoTime();
}
```
and it didn't do the trick - CPU got to 100%. I'll try to remove `@Substitute` if that helps, but my undestanding is probably still incorrect. I should really [RTFM](https://quarkus.io/guides/extension-authors-guide).

Reply:
Didn't help.

Reply:
I don't think this will work, the aliased field will still be initialized at build time, so will have the same value. If you want to use RecomputeFieldValue I think you need to use 'Custom' and then provide a CustomFieldValueComputer that returns the time.

Reply:
Ok, I've tried the `CustomFieldValueComputer` and besides the nasty system-path dependency to `GRAALVM_HOME/jre/lib/jvmci/jvmci-api.jar` it doesn't work. It seemed to me that actually the cleanest way would be
```java
    @BuildStep
    public RuntimeReinitializedClassBuildItem reinitScheduledFutureTask() {
        return new RuntimeReinitializedClassBuildItem(io.netty.util.concurrent.ScheduledFutureTask.class.getName());
    }
```
but for some reason this doesn't do the trick either - is there a way to verify that this is being applied?

Reply:
The `FromAlias` won't work if the original field value was constant, so both the original *and* the recompute alias have to be assigned to `nanoTime()`.  In fact if you have access to the original, you can just put the annotation right on there IIRC.

Reply:
@dmlloyd Do you mean constant as `static final` or in the sense of compile-time-constant? Because it's the former, set to `System.nanoTime()`:
https://github.com/netty/netty/blob/4.1/common/src/main/java/io/netty/util/concurrent/ScheduledFutureTask.java#L29
and the code excerpt using `@FromAlias` above is setting that as well.

Reply:
I meant compile time constant.  But if that's already the case then I guess that wasn't the issue.

I am very curious to work out why this didn't work in this case.  Maybe because it's a primitive value?

Reply:
@dmlloyd I've updated the PR as per your suggestion.